tools/Makefile

```diff
@@ -84,19 +46,39 @@ VPATH := $(filter-out -I,$(INCLUDES))
 .PHONY: all allopt opt.opt # allopt and opt.opt are synonyms
 allopt: opt.opt
 
+programs_byte := \
+  ocamldep ocamlprof ocamlcp ocamloptp ocamlmklib  \
+  ocamlmktop ocamlcmt dumpobj ocamlobjinfo \
+  primreq stripdebug cmpbyt
+install_files += $(filter $(installed_tools), $(programs_byte))
+programs_opt := $(programs_byte:%=%.opt)
+
+programs_byte_exe := $(programs_byte:%=%$(EXE))
+programs_opt_exe := $(programs_opt:%=%$(EXE))
+$(programs_byte) $(programs_opt): %: $(if $(EXE), %$(EXE), )
+
+$(programs_byte_exe):
+	$(CAMLC) $(LINKFLAGS) -I $(ROOTDIR) -o $@ $(filter-out %.cmi,$^)
+
+$(programs_opt_exe):
+	$(CAMLOPT_CMD) $(LINKFLAGS) -I $(ROOTDIR) -o $@ $(filter-out %.cmi,$^)
+
+all: $(programs_byte_exe)
+opt.opt: $(programs_opt_exe)
+clean::
+	rm -f $(programs_byte_exe) $(programs_opt_exe)
```

Comment:
This isn't quite right - the `clean` target can't depend `$(EXE)`. `clean` should _always_ erase `ocamldep`, ocamldep.opt`, `ocamldep.exe` and `ocamldep.opt.exe` (i.e. 4 files).

Reply:
Fixed. (I had thought the old behaviour was a mistake)

Reply:
The reason is that people may want to clean in a not-yet-configured or broken-configuration state. (Maybe we should have a comment to point that out.)